extensions/amp-story/1.0/test/test-analytics.js

```diff
@@ -53,4 +58,42 @@ describes.fakeWin('amp-story analytics', {}, env => {
 
     expect(trigger).to.have.been.calledWith('story-last-page-visible');
   });
+
+  it('should not mark an event as repeated the first time it fires', () => {
+    const trigger = sandbox.spy(analytics, 'triggerEvent');
+
+    storeService.dispatch(Action.CHANGE_PAGE, {
+      id: 'test-page',
+      index: 1,
+    });
+
+    expect(trigger).to.have.been.calledWith('story-page-visible');
+    expect(analytics.getVariableDetailsForTesting()).to.deep.include({
+      'detailsForPage': {},
+    });
+  });
+
+  it('should mark event as repeated when fired more than once', () => {
+    const trigger = sandbox.spy(analytics, 'triggerEvent');
+
+    storeService.dispatch(Action.CHANGE_PAGE, {
+      id: 'test-page',
+      index: 1,
+    });
+
+    storeService.dispatch(Action.CHANGE_PAGE, {
+      id: 'test-page2',
+      index: 2,
+    });
+
+    storeService.dispatch(Action.CHANGE_PAGE, {
+      id: 'test-page',
+      index: 1,
+    });
+
+    expect(trigger).to.have.been.calledWith('story-page-visible');
```

Comment:
Nit: `calledTwiceWith`?

Reply:
hmm I don't think sinon has this? i added `calledThrice` but lmk if you prefer another of the [spy properties](https://sinonjs.org/releases/latest/spies/)